includes/class-webmention-receiver.php

```diff
@@ -711,24 +715,66 @@ public static function publish_post_hook( $post_id ) {
 	 * @param WP_Error $error
 	 */
 	public static function delete( $error ) {
-		$error_codes = apply_filters( 'webmention_supported_delete_codes', array(
-			'resource_not_found',
-			'resource_deleted',
-			'resource_removed',
-		) );
+		$error_codes = apply_filters(
+			'webmention_supported_delete_codes', array(
+				'resource_not_found',
+				'resource_deleted',
+				'resource_removed',
+			)
+		);
 
 		if ( ! in_array( $error->get_error_code(), $error_codes ) ) {
 			return;
 		}
 
 		$commentdata = $error->get_error_data();
-		$commentdata = self::check_dupes($commentdata);
+		$commentdata = self::check_dupes( $commentdata );
 
 		if ( isset( $commentdata['comment_ID'] ) ) {
 			wp_delete_comment( $commentdata['comment_ID'] );
 		}
 	}
 
+
+	public static function whitelist_approved( $commentdata ) {
+		if ( ! $commentdata || is_wp_error( $commentdata ) ) {
+			return $commentdata;
+		}
+		if ( self::whitelist_check( $commentdata['source'] ) ) {
+			$commentdata['comment_approved'] = 1;
+		}
+		return $commentdata;
+	}
+
+	/**
+	 *
+	 * @param array $author_url
+	 *
+	 * @return boolean
+	 *
+	 */
+	public static function whitelist_check( $url ) {
+		$whitelist = get_option( 'whitelist_domains' );
+		$whitelist = trim( $whitelist );
+		$host      = wp_parse_url( $url, PHP_URL_HOST );
+		// strip leading www, if any
+		$host = preg_replace( '/^www\./', '', $host );
+		if ( '' === $whitelist ) {
+			return false;
+		}
+		$domains = explode( ',', $whitelist );
```

Comment:
the blacklist/whitelists of WordPress are using "one item per line" instead of comma seperated, so we should be consistent.